charts/consul/templates/mesh-gateway-deployment.yaml

```
@@ -213,6 +213,8 @@ spec:
             {{- end }}
             {{- if and .Values.externalServers.enabled .Values.externalServers.tlsServerName }}
             -tls-server-name={{.Values.externalServers.tlsServerName }} \
+            {{- else if .Values.global.cloud.enabled }}
+            -consul-tls-server-name=server.{{ .Values.global.datacenter}}.{{ .Values.global.domain}} \
```

```suggestion
            -tls-server-name=server.{{ .Values.global.datacenter}}.{{ .Values.global.domain}} \
```
I believe this is `-tls-server-name` ?

---

this has been fixed.